### ISSUE TITLE
Unable to install module for EE with 2.9.X version

### ISSUE THREAD

**Issue by TdAmi** [Smile-SA/elasticsuite#1893]
Cannot install "Elasticsuite Shared Catalog" with ElasticSuite 2.9.0

### Preconditions

Magento Version : Magento EE 2.3.5-p1

ElasticSuite Version : 2.9.0

Environment : Developer

Third party modules : None

### Steps to reproduce

1. Run `composer require smile/module-elasticsuite-shared-catalog` as documentation said

### Expected result

1. Install ElasticSuite correctly

### Actual result

1. [Screenshot, logs]
`Problem 1
    - Installation request for smile/module-elasticsuite-shared-catalog ^1.1 -> satisfiable by smile/module-elasticsuite-shared-catalog[1.1.0].
    - smile/module-elasticsuite-shared-catalog 1.1.0 requires smile/elasticsuite ~2.8.0 -> satisfiable by smile/elasticsuite[2.8.0, 2.8.1, 2.8.2, 2.8.3, 2.8.4, 2.8.5, 2.8.6, 2.8.7, 2.8.8, 2.8.x-dev] but these conflict with your requirements or minimum-stability.`

If trying to use 2.8.X `composer require smile/elasticsuite:2.8.0` got this :
`Problem 1
    - Installation request for smile/elasticsuite 2.8.0 -> satisfiable by smile/elasticsuite[2.8.0].
    - Conclusion: remove elasticsearch/elasticsearch v7.8.0
    - Conclusion: don't install elasticsearch/elasticsearch v7.8.0
    - smile/elasticsuite 2.8.0 requires elasticsearch/elasticsearch ~5.1|~6.1 -> satisfiable by elasticsearch/elasticsearch[v5.1.0, v5.1.1, v5.1.2, v5.1.3, v5.2.0, v5.3.0, v5.3.1, v5.3.2, v5.4.0, v5.5.0, v6.1.0, v6.5.0, v6.5.1, v6.7.0, v6.7.1, v6.7.2].
    - Can only install one of: elasticsearch/elasticsearch[v5.1.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.1.1, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.1.2, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.1.3, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.2.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.3.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.3.1, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.3.2, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.4.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v5.5.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.1.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.5.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.5.1, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.7.0, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.7.1, v7.8.0].
    - Can only install one of: elasticsearch/elasticsearch[v6.7.2, v7.8.0].
    - Installation request for elasticsearch/elasticsearch (locked at v7.8.0) -> satisfiable by elasticsearch/elasticsearch[v7.8.0].`

**Comment by vahonc** [comment]
Hi @TdAmi,

I tried to reproduce your issue with the Magento 2.3.5-p2 EE version.
It looks like currently, we have some problem linked to requirements for smile/module-elasticsuite-shared-catalog and smile/elasticsuite 2.9.x

I also tried to reproduce this issue using smile/elasticsuite 2.8.x and got just like you this:
`composer require smile/elasticsuite ~2.8.0`

Problem 1
    - Conclusion: don't install smile/elasticsuite 2.8.9
    - Conclusion: don't install smile/elasticsuite 2.8.8
    - Conclusion: don't install smile/elasticsuite 2.8.7
    - Conclusion: don't install smile/elasticsuite 2.8.6
    - Conclusion: don't install smile/elasticsuite 2.8.5
    - Conclusion: don't install smile/elasticsuite 2.8.4
    - Conclusion: don't install smile/elasticsuite 2.8.3
    - Conclusion: don't install smile/elasticsuite 2.8.2
    - Conclusion: don't install smile/elasticsuite 2.8.1
    - Conclusion: remove elasticsearch/elasticsearch v7.9.0

I checked version of elasticsearch/elasticsearch module:    
`composer show elasticsearch/elasticsearch => v7.9.0`

Therefore, as a temporary solution, of course if it will suitable for you, you can try to downgrading elasticsearch/elasticsearch module from v7.9.0 to v6.7.2 like this:

`composer remove elasticsearch/elasticsearch`
`composer require elasticsearch/elasticsearch ~6.7.0`

Installing smile/elasticsuite module:
`composer require smile/elasticsuite ~2.8.0`

And then installing smile/module-elasticsuite-shared-catalog module:
`composer require smile/module-elasticsuite-shared-catalog`

We come back to the issue linked to requirements for smile/module-elasticsuite-shared-catalog and smile/elasticsuite 2.9.x on next week.

Best regards

**Comment by rbayet** [comment]
Hello @TdAmi,

I'm closing this issue since
* @romainruaud  made a new [1.1.1](https://github.com/Smile-SA/magento2-module-elasticsuite-shared-catalog/releases/tag/1.1.1) release of Elasticsuite Shared Catalog in August allowing using both Elasticsuite 2.8.x and 2.9.x
* we also introduced a new branch/release strategy two weeks ago with version numbering similar to those of Elasticsuite
  * there is now a 2.8.x branch, a 2.9.x branch and a 2.10.x branch for Elasticsuite Shared Catalog
  * long story short, if you installed Elasticsuite with "composer require smile/elasticsuite ~2.9.0", then you just need to install Elasticsuite Shared Catalog with "composer require smile/module-elasticsuite-shared-catalog ~2.9.0"

See the **Which version should I use ?** section in [Elasticsuite Shared Catalog README.md](https://github.com/Smile-SA/elasticsuite/blob/2.10.x/README.md)

Regards,
